splinterd/src/config/builder.rs

```diff
@@ -12,14 +12,441 @@
+        if let Some(storage) = partial.storage() {
+            self.with_storage(source.to_owned(), storage);
+        }
+        if let Some(transport) = partial.transport() {
+            self.with_transport(source.to_owned(), transport);
+        }
+        if let Some(cert_dir) = partial.cert_dir() {
+            self.with_cert_dir(source.to_owned(), cert_dir);
+        }
+        if let Some(ca_certs) = partial.ca_certs() {
+            self.with_ca_certs(source.to_owned(), ca_certs);
+        }
+        if let Some(client_cert) = partial.client_cert() {
+            self.with_client_cert(source.to_owned(), client_cert);
+        }
+        if let Some(client_key) = partial.client_key() {
+            self.with_client_key(source.to_owned(), client_key);
+        }
+        if let Some(server_cert) = partial.server_cert() {
+            self.with_server_cert(source.to_owned(), server_cert);
+        }
+        if let Some(server_key) = partial.server_key() {
+            self.with_server_key(source.to_owned(), server_key);
+        }
+        if let Some(service_endpoint) = partial.service_endpoint() {
+            self.with_service_endpoint(source.to_owned(), service_endpoint);
+        }
+        if let Some(network_endpoint) = partial.network_endpoint() {
+            self.with_network_endpoint(source.to_owned(), network_endpoint);
+        }
+        if let Some(peers) = partial.peers() {
+            self.with_peers(source.to_owned(), peers);
+        }
+        if let Some(node_id) = partial.node_id() {
+            self.with_node_id(source.to_owned(), node_id);
+        }
+        if let Some(bind) = partial.bind() {
+            self.with_bind(source.to_owned(), bind);
+        }
+        #[cfg(feature = "database")]
+        {
+            if let Some(database) = partial.database() {
+                self.with_database(source.to_owned(), database);
+            }
+        }
+        if let Some(registry_backend) = partial.registry_backend() {
+            self.with_registry_backend(source.to_owned(), registry_backend);
+        }
+        if let Some(registry_file) = partial.registry_file() {
+            self.with_registry_file(source.to_owned(), registry_file);
+        }
+        if let Some(heartbeat_interval) = partial.heartbeat_interval() {
+            self.with_heartbeat_interval(source.to_owned(), heartbeat_interval);
+        }
+        if let Some(timeout) = partial.admin_service_coordinator_timeout() {
+            self.with_admin_service_coordinator_timeout(source.to_owned(), timeout);
+        }
+        if let Some(state_dir) = partial.state_dir() {
+            self.with_state_dir(source.to_owned(), state_dir);
+        }
+    }
+
+    /// Adds a `storage` value to the ConfigBuilder object.
+    ///
+    /// # Arguments
+    ///
+    /// * `storage` - The type of storage that should be used to store circuit state.
+    ///
+    pub fn with_storage(&mut self, source: ConfigSource, storage: String) {
```

Comment:
Can these with_* statements be private? Seems like we would only need with_partial_config() to be public. Seems like they could be inlined in the above code where they are used.

Reply:
Addressed, these methods were removed.